lib/Backend/GlobOpt.cpp

```diff
@@ -17187,6 +17187,13 @@ GlobOpt::PRE::InsertSymDefinitionInLandingPad(StackSym * sym, Loop * loop, Sym *
 
             BasicBlock* loopTail = loop->GetAnyTailBlock();
             Value * valueOnBackEdge = loopTail->globOptData.FindValue(propSym);
+            
+            // If o.x is not invariant in the loop, we can't use the preloaded value of o.x.y in the landing pad
+            Value * valueInLandingPad = loop->landingPad->globOptData.FindValue(propSym);
+            if (valueOnBackEdge->GetValueNumber() != valueInLandingPad->GetValueNumber())
```

Comment:
>valueInLandingPad [](start = 53, length = 17)

Is valueInLandingPad guaranteed not to be null here?

Reply:
Yes. If the `propSym` is a PRE candidate, then it must have a value on the back edge

---
In reply to: [200791202](https://github.com/Microsoft/ChakraCore/pull/5420#discussion_r200791202) [](ancestors = 200791202)

Reply:
Oh, you were asking about `valueInLandingPad`. It should also be non-null if `propSym` is a pre candidate - either `propSym` was live before the loop, or we must have added an initial value for it in the landing pad.

---
In reply to: [201124558](https://github.com/Microsoft/ChakraCore/pull/5420#discussion_r201124558) [](ancestors = 201124558,200791202)